### PR TITLE
Release v0.4.0

### PR DESCRIPTION
**Pull request by ShivanshVij**

### CODE REVIEW

File: module.ts [hunk 1]
```diff
@@ -45,7 +48,21 @@ export class Module<T extends Signature> {
...
     constructor(template: Template<T>) {
         this.template = template;
-        this.wasi = new DisabledWASI(this.template.env, this.template.runtime.config.stdout, this.template.runtime.config.stderr);
+
+        const name = `${this.template.identifier}.${uuid()}`
+
+        let stdout = this.template.runtime.config.stdout;
+        let stderr = this.template.runtime.config.stderr;
+
+        if (stdout !== undefined && !this.template.runtime.config.rawOutput) {
```
Comment (SuperManifolds) [inline module.ts:57]: It is discouraged to make direct comparisons with undefined as there are multiple situations in JS where this has unintended consequences (more info here https://eslint.org/docs/latest/rules/no-undefined)
I would rewrite this as `if (typeof stdout !== "undefined"`

**Review by SuperManifolds — CHANGES_REQUESTED** [review]

File: module.ts (hunk 1 above)
Comment (ShivanshVij) [reply module.ts:57]: This isn't the only place this comparison is occurring. I'll change it everywhere then.

Comment (ShivanshVij) [reply module.ts:57]: Screenshot 2023-09-20 at 11 52 03 AM

Looks like doing it with `typeof` does not satisfy the typescript compiler.

Comment (SuperManifolds) [reply module.ts:57]: the result of typeof is a string value, you must compare with `"undefined"`

Comment (ShivanshVij) [reply module.ts:57]: Screenshot 2023-09-20 at 11 53 48 AM

Should I be making the change for the `===` side as well?

Comment (SuperManifolds) [reply module.ts:57]: Yes

Comment (ShivanshVij) [reply module.ts:57]: Screenshot 2023-09-20 at 11 57 15 AM

I should modify the signature templates as well right?

Comment (SuperManifolds) [reply module.ts:57]: Indeed

**Review by SuperManifolds — APPROVED** [review]
LGTM

**Pull request merged**